src/Microsoft.Health.Fhir.Core/Features/Context/CosmosStorageContext.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Health.Fhir.Core.Features.Context
+{
+    /// <summary>
+    /// A Mediatr notification containing statistics about Cosmos DB usage while handling a request.
+    /// This gets emitted by the ApiNotificationMiddleware when a response is returned by the server.
+    /// Consume these using Mediatr to collect stats about Cosmos DB usage.
+    /// </summary>
+    public class CosmosStorageContext : IStorageContext
```

Comment:
I think the interface could be something like `IStorageRequestStatistics` and this class `CosmosStorageRequestStatistics` or similiar. Maybe metrics instead of statistics.